### ISSUE TITLE
behavior: closing multi-point shapes

### ISSUE THREAD

**Issue by dwelle** [excalidraw/excalidraw#2035]
Currently when you move/create last point close to the first (~10px), we auto-close the shape. This is not a real point merge, the two points are still retained separately.

This has following behavior:

1. It creates a sharp corner:

   ![image](https://user-images.githubusercontent.com/5153846/90244714-9bfc7a00-de31-11ea-8310-065f5507068a.png)

2. During edit, moving one of the terminal points does not move the other one:

	![excal_drag_terminal_point](https://user-images.githubusercontent.com/5153846/90244783-bd5d6600-de31-11ea-8c08-6f5fe4641d17.gif)

Both of these could be considered bugs or features. 

For example, (1) is good for creating shapes like the above heart. But it's bad for creating other shapes where you want to make all corners round.

(2) can be good for disconnecting the two shapes, but bad when you try to move them both. On the whole I think (2) should be fixed, and the ability to disconnect would be done simply by deleting the point.

(1) can be "fixed" by physically merging the two points instead of just overlaying them across one another, but we'd then lose the ability for sharp corner. Or it can be fixed by implementing a separate "fully-round" sharpness option as [suggested](https://github.com/excalidraw/excalidraw/pull/1931#issuecomment-674017879) by @dai-shi.

**Comment by dai-shi** [comment]
This might be out of the scope of this issue, but for me it's somewhat related.
Eventually, we may want to support any combination of sharp and round corners.

For example, a rectangle with two sharp corners and two round corners.

<img width="473" alt="image" src="https://user-images.githubusercontent.com/490574/90249403-5cf01800-de75-11ea-9e38-4ea14b71307b.png">

I can't think of any good UI for this though, except for something like [bezier handles](https://github.com/excalidraw/excalidraw/issues/1605#issuecomment-666731276).

**Comment by dwelle** [comment]
I agree it'd a nice feature. That said, it may fall under the category of "too advanced, won't implement".

One way to do it without bezier handles is to have per-point configs ala Figma. It'd like be part of a floating menu if we ever get to implement that.

**Comment by dai-shi** [comment]
Makes sense. Thanks for the clarification.

Me:
> we'd stick with import-only svg path (not editable).

👉 #1924

**Comment by timcolson** [comment]
TIL that I can overlap the first and last points to "close" a shape. Yay!!! 😄 
...300 ms later...
Wait, I can't move them as one or make them smooth/round? 😢 
But, but, but, how do all those custom shapes do it? 
...4.2 min later...
<finds this issue> 
<reads it>
<checks last update date...>
Bummer. 😭

**Comment by dwelle** [comment]
@timcolson how do you feel about https://github.com/excalidraw/excalidraw/issues/4359?

**Comment by timcolson** [comment]
#4359 is intriguing! Essentially it is "closing" a shape, if fill color is set.  
(FYI - Acorn appears to take this approach too. But it also can truly "close" the last points
<img width="1198" alt="acorn_screen" src="https://user-images.githubusercontent.com/1376728/191396476-6d043b70-4feb-49a1-a95a-e5ed8a9e0f6d.png">

Today, I realized how two "rounded" edges in Excalidraw, placed super close, form a sharp edge. (ex. tip of the heart). 

I also learned that ...
* **all points** in a shape are either round or sharp.
*  edges cannot be added to built-in shapes. (ex. start with rect, pull out one more edge to make a pentagon) 

I appreciate the creativity of #4359. Backwards compatibility could be problematic. I imagine many surprises, and not the good kind, for existing diagrams.

TL;DR - I don't know that #4359 would solve my need to have "all rounded" for a custom shape. 

I experimented with drawing filled, rounded shapes in other tools: Acorn, Inkscape, Omnigraffle, and Figma. 
A common approach requires each point to be part of a Bezier path, controlled individually. Handles can make the point rounded or angular. 

For quick, rough drawings, and collaboration, Excalidraw has no equal! For this specific use case, one of the more complex tools seems a better fit for now. I look forward to seeing how closed shapes evolve in Excalidraw! Thx!

**Comment by TehBrian** [comment]
I think that, conceptually, closed shapes should either 1) allow all points to be set to rounded/straight, or 2) allow points to be individually set to rounded/straight. If 2) is deemed "too advanced" for Excalidraw (personally, I don't think it is), then I think we should go with 1).

The current model model is a weird mix of 1) and 2), which is: "either all the points are straight *or* all the points are rounded—except for the first and last point." This doesn't feel like a conscious design decision that benefits users; rather, it makes the current merging implementation feel unfinished. (It's why I searched for this issue as a user of the Obsidian Excalidraw plugin.)

I don't mean to rush development, but I feel like this issue should be considered somewhat high priority as it currently forces all rounded polygons to have a weird jagged corner.

**Comment by TehBrian** [comment]
Also, I've personally run into one use-case for allowing points to be individually set to rounded/straight: sketching segments of 3D surfaces.

<img width="467" height="389" alt="Image" src="https://github.com/user-attachments/assets/46ec0f4a-6f81-4f1e-ae11-f7e15ee12d7e" />

I think it'd be reasonable and user-friendly to:
- keep the buttons for setting vertex smoothness,
- have them that reset the smoothness of all points,
- and allow users to individually set point smoothness.

> It'd likely be part of a floating menu if we ever get to implement that.

It's been a while since you said that, @dwelle, but I'm assuming you were referring to the dialog upon right clicking points? Or did you mean a separate floating dialog that appears as soon as a point is selected/highlighted?
